docs: document ESLint migration codemods in v9 and v10 guides (#20980)
eslint/eslint: docs/src/use/migrate-to-10.0.0.md

    order: 9
---

> {%- from 'components/npx_tabs.macro.html' import npx_tabs %}

Reviewer: We need to remove `{%- from 'components/npx_tabs.macro.html' import npx_tabs %}`

Reviewer: I believe, this is unused.

ESLint v10.0.0 is a major release of ESLint, and as such, has several breaking changes that you need to be aware of. This guide is intended to walk you through the breaking changes.

> To help with this migration, ESLint provides migration codemods to automate many of the changes described in this guide. All official ESLint codemods are available in the [eslint/codemods](https://github.com/eslint/codemods) repository and through the [Codemod Registry](https://codemod.link/eslint).

Reviewer 2: ```suggestion
To help with this migration, ESLint provides migration codemods to automate many of the changes described in this guide. All official ESLint codemods are available in the [eslint/codemods](https://github.com/eslint/codemods) repository and through the [Codemod Registry](https://app.codemod.com/registry?q=scope%3Aeslint).
```

The `https://codemod.link/eslint` link redirects to `https://app.codemod.com/registry?q=scope%3Aeslint`, so maybe the latter is more appropriate here?

Author: it was intentional (for link analytics and easier updates on our end), but since you mentioned it, I'm just going to use the direct link.

## Use migration codemods

The `@eslint/v9-to-v10` codemod upgrades ESLint projects from v9 to v10. It includes four individual codemods that can also be run independently:

- `@eslint/v9-to-v10-config`: Remove legacy env vars and CLI flags
- `@eslint/v9-to-v10-custom-rules`: Replace removed `context` and `SourceCode` methods
- `@eslint/v9-to-v10-ruletester`: Clean up `RuleTester` test cases
- `@eslint/v9-to-v10-linter-api`: Fix `Linter`/`ESLint` API usage

```shell
npx codemod @eslint/v9-to-v10
```

> Learn more in the [Codemod Registry](https://codemod.link/eslint).

Reviewer 2: ```suggestion
Learn more in the [Codemod Registry](https://app.codemod.com/registry?q=scope%3Aeslint).
```

Codemods are a starting point. Review the changes and consult the breaking changes below for anything the codemods do not cover.

The lists below are ordered roughly by the number of users each change is expected to affect, where the first items are expected to affect the most users.

## Table of Contents

> - [Use migration codemods](#use-migration-codemods)

Reviewer 2: ```suggestion

```

I’m not sure this TOC should be located here, since TOCs are usually at the top level before the main topics. I think it would be fine to remove it so that the table of contents focuses only on breaking-change-related topics.

Or, another approach would be to place `## Use migration codemods` after the TOC section so that users only scroll down from the TOC.

- Node.js v22.13.0 and above
- Node.js v24 and above

> **Codemod:** The migration codemods do not cover this change.

Reviewer 2: ```suggestion
**Codemod:** The migration codemods do not cover this change.

```

Non-blocking suggestion, but just to maintain the same format.

- Node.js v22.13.0 and above
- Node.js v24 and above

> **Codemod:** The migration codemods do not cover this change.

Reviewer 2: ```suggestion
```

It looks like in https://github.com/eslint/eslint/pull/20980/changes/208cfa78db90331d8aad642f0136e18181e9ab52, the `**Codemod:** The migration codemods do not cover this change.` sections were removed entirely.

Could we also remove these sections in `docs/src/use/migrate-to-10.0.0.md` for consistency? (The same applies to the entire documentation.)

Author: done. thanks @lumirlumir

- Node.js v22.13.0 and above
- Node.js v24 and above

> **To address:** The migration codemods do not cover this change. Make sure you upgrade to at least Node.js v20.19.0 when using ESLint v10.0.0. One important thing to double check is the Node.js version supported by your editor when using ESLint via editor integrations. If you are unable to upgrade, we recommend continuing to use ESLint v9 until you are able to upgrade Node.js.

Reviewer 2: ```suggestion
**Codemod:** The migration codemods do not cover this change. 
**To address:** Make sure you upgrade to at least Node.js v20.19.0 when using ESLint v10.0.0. One important thing to double check is the Node.js version supported by your editor when using ESLint via editor integrations. If you are unable to upgrade, we recommend continuing to use ESLint v9 until you are able to upgrade Node.js.
```

To separate concerns, I think a separate Codemod section would be preferable here. And it applies to the other sections as well.

- [`no-useless-assignment`](../rules/no-useless-assignment)
- [`preserve-caught-error`](../rules/preserve-caught-error)

> **Codemod:** The migration codemods do not cover this change.

Reviewer 3: If the first word refers to the migration tool, it should be written as `codemod`. If this is a quotation, it should be formatted as a blockquote, like this:

> Codemod: test
